src/compat/zygote.jl

```diff
@@ -193,3 +193,126 @@ end
         return replace_diag(log, Y)
     end
 end
+
+@adjoint function inv_link_w_lkj(y)
+    K = size(y, 1)
+
+    z = tanh.(y)
+    w = similar(z)
+    
+    w[1,1] = 1
+    for j in 1:K
+        w[1, j] = 1
+    end
+
+    for i in 2:K
+        for j in 1:(i-1)
+            w[i, j] = 0
+        end
+        for j in i:K
+            w[i, j] = w[i-1, j] * sqrt(1 - z[i-1, j]^2)
+        end
+    end
+
+    w1 = copy(w) # cache result
+
+    for i in 1:K
+        for j in (i+1):K
+            w[i, j] = w[i, j] * z[i, j]
+        end
+    end
+
+    return w, Δw -> begin
+        Δz = zeros(size(Δw))
+        Δw1 = zeros(size(Δw))
+        for i in 1:K, j in (i+1):K
+            Δw1[i,j] = Δw[i,j] * z[i,j]
+            Δz[i,j] = Δw[i,j] * w1[i,j]
+        end
+        for i in 1:K
+            Δw1[i,i] = Δw[i,i]
+        end
+
+        for j=2:K, i=j:-1:2
+            tz = sqrt(1 - z[i-1, j]^2)
+            Δw1[i-1, j] += Δw1[i, j] * tz
+            Δz[i-1, j] += Δw1[i, j] * w1[i-1, j] * 0.5 / tz * (-2 * z[i-1, j])
+        end
+
+        Δy = Δz .* (1 ./ cosh.(y).^2)
+        return (Δy,)
+    end
+end
+
+@adjoint function link_w_lkj(w)
+    # println("link_w_lkj $(typeof(w) == Matrix{Float64})")
+    K = size(w, 1)
+    z = zero(w)
+    
+    for j=2:K
+        z[1, j] = w[1, j]
+    end
+
+    #=
+    for i=2:K, j=(i+1):K
+        z[i, j] = w[i, j] / w[i-1, j] * z[i-1, j] / sqrt(1 - z[i-1, j]^2)
+    end
+    =#
```

Comment:
I guess this is the same as before? The code below is an "alternative" impl of this commented out section?

Reply:
Yes.